subsys/net/lib/websocket/websocket.c

```diff
@@ -443,12 +443,25 @@ static int websocket_ioctl_vmeth(void *obj, unsigned int request, va_list args)
 }
 
 static void websocket_mask_payload(u8_t *payload, size_t payload_len,
-				   u32_t masking_value)
+				   u32_t masking_value, s8_t mask_shift)
 {
-	int i;
+	int *mask_shift_ptr;
+	int i, zero = 0;
+
+	/* Special handling when receiving data. As we cannot guarantee that
+	 * read data is received in four byte blocks, we need to pick proper
+	 * byte from the shift mask. When sending data, the mask_shift is set
+	 * to -1.
+	 */
+	if (mask_shift < 0) {
+		mask_shift_ptr = &zero;
+	} else {
+		mask_shift_ptr = &mask_shift;
```

Comment:
How var of type `int*` can be assigned with address of `s8_t`?

Reply:
Ah, simple typo error. I will refactor this function or probably remove it altogether as it is now looking quite convoluted.